src/js/pages/Index.js

```diff
@@ -144,21 +138,34 @@ var Index = React.createClass({
     );
   },
 
-  handleWindowResize(event) {
-    const windowWidth = event.target.innerWidth;
-    const { isVisible } = this.internalStorage_get();
+  handleWindowResize() {
+    const currentWindowWidth = window.innerWidth;
 
-    if (windowWidth <= MOBILE_THRESHOLD && isVisible) {
-      SidebarActions.close();
+    if (
+      !hasViewportChanged(this.state.previousWindowWidth, currentWindowWidth)
+    ) {
+      this.setState({
+        previousWindowWidth: currentWindowWidth
+      });
+
+      return;
     }
 
-    if (windowWidth > MOBILE_THRESHOLD && !isVisible) {
+    if (getCurrentViewport() === viewport.DESKTOP) {
       SidebarActions.open();
     }
+
+    if (getCurrentViewport() === viewport.MOBILE) {
+      SidebarActions.close();
+    }
+
+    this.setState({
```

Comment:
this can be moved into the if clauses 🤔

Reply:
Not sure why I would do that if I will need to update every time there is a change.

Reply:
yeah, you're right… 😅